class-parsely-recommended-widget.php

```diff
@@ -359,4 +365,25 @@ private function get_boost_params() {
 			'pi_referrals'          => __( 'Page views where the referrer was pinterest.com', 'wp-parsely' ),
 		);
 	}
+
+	private function api_key_and_secret_are_populated() {
+		$options = get_option( 'parsely' );
+
+		// No options are saved, so API key is not available.
+		if ( ! is_array( $options ) ) {
+			return false;
+		}
+
+		// Parse.ly Site ID settings field is not populated.
+		if ( ! array_key_exists( 'apikey', $options ) || $options['apikey'] === '' ) {
+			return false;
+		}
+
+		// Parse.ly API Secret settings field is not populated.
+		if ( ! array_key_exists( 'api_secret', $options ) || $options['api_secret'] === '' ) {
+			return false;
+		}
```

Comment:
Unless we need the specificity of checking for an empty string, we can simplify this by consulting the `empty` language construct & save a few bytes:

```suggestion
		if ( empty( $options['apikey'] ) ) {
			return false;
		}

		// Parse.ly API Secret settings field is not populated.
		if ( empty( $options['api_secret'] ) ) {
			return false;
		}
```

Even with error reporting set to `E_ALL` it won't generate a warning or notice if the variable or key is not set.

Reply:
I would favour clarity of intent with the code, than saving a few bytes.

https://beberlei.de/2021/02/19/when_to_use_empty_in_php_i_say_never.html

Reply:
Agree there are lots of times when empty is problematic, but IMO this is one of those cases where it's suited (checking that the variable & key are set and the value isn't empty. We're not overly concerned w/ data types as it's decidedly a string (stored in options, [validated as string](https://github.com/Parsely/wp-parsely/blob/1a398fac2ebc4935199299977522c5a817b62505/wp-parsely.php#L596)).

That said, I don't feel strongly about the suggestion, so feel free to leave as is if you feel it adds clarity or safety.